proxy/healthy_endpoints_test.go

```diff
@@ -128,63 +151,134 @@ func TestPHCForSingleHealthyEndpoint(t *testing.T) {
 }
 
 func TestPHCForMultipleHealthyEndpoints(t *testing.T) {
-	services := []*httptest.Server{}
-	for i := 0; i < 3; i++ {
-		service := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(http.StatusOK)
-		}))
-		services = append(services, service)
-		defer service.Close()
-	}
+	services := setupServices(t, 3, 0)
 
 	for _, algorithm := range []string{"random", "consistentHash", "roundRobin", "powerOfRandomNChoices"} {
 		t.Run(algorithm, func(t *testing.T) {
 			_, ps := setupProxy(t, fmt.Sprintf(`* -> consistentHashKey("${request.header.ConsistentHashKey}") -> <%s, "%s", "%s", "%s">`,
-				algorithm, services[0].URL, services[1].URL, services[2].URL))
-			failedReqs := sendGetRequests(t, ps)
-			assert.Equal(t, 0, failedReqs)
+				algorithm, services[0], services[1], services[2]))
+			va := fireVegeta(t, ps, 3000, 1*time.Second, 5*time.Second)
+			count200, ok := va.CountStatus(200)
+			assert.True(t, ok)
+			assert.Equal(t, count200, int(va.TotalRequests()))
```

Comment:
you should also check that the amount is not 0 or maybe >=N for an N that you expect